server.js

```diff
@@ -1,38 +1,81 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+
+import topMusicData from "./data/top-music.json";
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+const Track = new mongoose.model("Track", {
+  trackName: String,
+  artistName: String,
+  genre: String,
+  danceability: Number,
+  popularity: Number,
+});
+
+if (process.env.RESET_DATABASE) {
+  const populateDatabase = async () => {
+    await Track.deleteMany();
+
+    topMusicData.forEach((item) => {
+      const newTrack = new Track(item);
+      newTrack.save();
+    });
+  };
+  populateDatabase();
+}
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send(
+    `Welcome to the Top Music API with ${topMusicData.length} listed tracks.`
+  );
+});
+
+app.get("/tracks", async (req, res) => {
+  const allTracks = await Track.find(req.query);
+  res.json(allTracks);
+});
+
+app.get("/track/:id", async (req, res) => {
+  const singleTrack = await Track.findById(id);
+  if (singleTrack) {
+    res.json(singleTrack);
+  } else {
+    res.status(404).json({ error: "Invalid track id" });
+  }
+});
+
+app.get("/artists/:artist/tracks", (req, res) => {
+  const { artist } = req.params;
+  const artistCapitalized = artist.charAt(0).toUpperCase() + artist.slice(1);
```

Comment:
Nice ⭐

Reply:
Would love some comments though, as I can't really see what you are accomplishing (as the API, unfortunately, does not work)